fontforge/python.c

```diff
@@ -1790,17 +1850,17 @@ return( reductionTuple );
 static int PyFFPoint_compare(PyFF_Point *self,PyObject *other) {
     double x, y;
 
-    /* I'd like to accept general sequences but there is no PyArg_ParseSequence*/
-    if ( PyTuple_Check(other) && PyTuple_Size(other)==2 ) {
-	if ( !PyArg_ParseTuple(other,"dd", &x, &y ))
-return( -1 );
-    } else if ( PyType_IsSubtype(&PyFF_PointType, Py_TYPE(other)) ) {
-	x = ((PyFF_Point *) other)->x;
-	y = ((PyFF_Point *) other)->y;
-    } else {
-	PyErr_Format(PyExc_TypeError, "Unexpected type");
-return( -1 );
+    if ( !PyArg_ParseTuple(other,"dd", &x, &y ) ) {
+	PyErr_Clear();
```

Comment:
So, just to be clear, the idea is to parse even if it generates an error and then to depend on the subtype check to determine validity?

Reply:
The existing code is basically redundant, except perhaps in the difference in error messages generated by `PyErr_Format()` and `ParseTuple()` (if the latter doesn't find elements it can convert to numbers.) The new code first tries to parse the tuple and if that fails checks if the argument is a point, and if that fails produces the error. This is a better match for most of the rest of the tuple parsing code in the file.